src/HttpClientFactory/Polly/test/PolicyHttpMessageHandlerTest.cs

```diff
@@ -147,6 +147,58 @@ public async Task SendAsync_StaticPolicy_PolicyTriggers_CanReexecuteSendAsync_Fi
             Assert.True(fakeContent.Disposed);
         }
 
+        [Fact]
+        public async Task MultipleHandlers_CanReexecuteSendAsync_FirstResponseDisposed()
+        {
+            // Arrange
+            var policy1 = HttpPolicyExtensions.HandleTransientHttpError()
+                .RetryAsync(retryCount: 1);
+            var policy2 = HttpPolicyExtensions.HandleTransientHttpError()
+                .CircuitBreakerAsync(handledEventsAllowedBeforeBreaking: 2, durationOfBreak: TimeSpan.FromSeconds(10));
+
+            var callCount = 0;
+            var fakeContent = new FakeContent();
+            var firstResponse = new HttpResponseMessage()
+            {
+                StatusCode = System.Net.HttpStatusCode.InternalServerError,
+                Content = fakeContent,
+            };
+            var expected = new HttpResponseMessage();
+
+            var handler1 = new PolicyHttpMessageHandler(policy1);
+            var handler2 = new PolicyHttpMessageHandler(policy2);
+            handler1.InnerHandler = handler2;
```

Comment:
do we need to explicitly do this mapping when register 2 policies in the httpClientBulider ? 
or just using 
 clientBuilder.AddPolicyHandler(first).AddPolicyHandler(second); will work?